kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/ExpressionUtils.java

```diff
@@ -174,4 +177,30 @@ static void compareBinary(ColumnVector left, ColumnVector right, int[] result) {
             }
         }
     }
+
+    /**
+     * Utility method to return the left child of the binary input expression
+     */
+    static Expression getLeft(Expression expression) {
```

Comment:
is there not a BinaryExpression class that could have a `getLeft` API?

This seems like a very c / struct type design, instead of a OOP design?

Reply:
im worried that these APIs won't be discoverable. with an IDE, i can see all of the APIs available for a given class. with this util organization, I won't be able to do that.

also, what's wrong with another class for just two methods? seems fine with me

not a blocker. we can decide and refactor later.

Reply:
There are not APIs, just utility methods in the expression evaluator which is a table client implementation. 

Anyways, if this becomes a problem for custom `TableClient` implementations, we can add the `BinaryExpression`.